### PR TITLE
Create CartItemView component, update ProductCard board

### PR DESCRIPTION
**Pull request by yurii-ve**
- created `CardItemView` component and used it in the board
- removed separate boards for ProductCard
- added 'Out of stock' and 'With discount' product cards to the "UI Kit - Components" board

### CODE REVIEW

File: src/components/cart/cart-item/cart-item-view.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    isLast?: boolean;
+    cartItem: cart.LineItem;
+    onRemoveButtonClick: () => void;
+    onQuantityChange: React.ChangeEventHandler<HTMLInputElement>;
+}
+
+export const CartItemView = ({
+    cartItem,
+    className,
+    isLast,
```
Comment (yurii-ve) [inline src/components/cart/cart-item/cart-item-view.tsx:21]: this is just moved from `CartItem` component... `isLast` is weird and should be removed in the next PR

File: _codux/boards/components/product-card/product-card.board.tsx [hunk 2]
```diff
@@ -1,15 +1,34 @@
...
     name: 'Product Card',
     Board: () => (
-        <ProductCard
-            name='Shel 50" Class LED 4K UHD Smart TV'
-            price={{ formatted: { price: '$85' } }}
-            style={{ width: '300px' }}
-            imageUrl="https://wixmp-b7f7090100b13623109851bc.wixmp.com/layouters-starters/img_02.jpg"
-        />
+        <div className={styles.root}>
+            <ProductCard
```
Comment (tetianaman) [inline _codux/boards/components/product-card/product-card.board.tsx]: Do you have `ProductCard` in UI-Kit board?

Comment (yurii-ve) [reply _codux/boards/components/product-card/product-card.board.tsx]: you're right, removed separate board and added 'Out of stock' and 'With discount' cards to the UI kit board.
Even though personally I don't like the idea :)

File: _codux/boards/ui-kits/ui-kit-components.board.tsx [hunk 3]
```diff
@@ -30,12 +30,31 @@ export default createBoard({
...
             </Variant>
-            <p className={styles.productCardInfo}>Product Card No Image</p>
+            <p className={styles.productCardInfo}>Product Card - No Image</p>
+            <Variant name="Product Card Without Image">
+                <ProductCard
+                    name="Im a product"
+                    imageUrl="https://wixmp-b7f7090100b13623109851bc.wixmp.com/layouters-starters/img_02.jpg"
+                    price={{ formatted: { price: '$15.00' } }}
+                    className={classNames(styles.productCard, styles.productCard)}
+                    outOfStock={true}
```
Comment (VladyslavG) [inline _codux/boards/ui-kits/ui-kit-components.board.tsx:42]: should the name of the variant specify out of stock ?

Comment (VladyslavG) [reply _codux/boards/ui-kits/ui-kit-components.board.tsx:42]: also it says no image, but I can see image url

Comment (VladyslavG) [reply _codux/boards/ui-kits/ui-kit-components.board.tsx:42]: is the name simply wrong ?

File: _codux/boards/ui-kits/ui-kit-components.board.tsx [hunk 4]
```diff
@@ -30,12 +30,31 @@ export default createBoard({
...
+                />
+            </Variant>
+            <p className={styles.productCardInfo}>Product Card - Out Of Stock</p>
+            <Variant name="Product Card Without Image">
```
Comment (VladyslavG) [inline _codux/boards/ui-kits/ui-kit-components.board.tsx]: same name as above, but still ... image is present

File: _codux/boards/ui-kits/ui-kit-components.board.tsx (hunk 3 above)
Comment (yurii-ve) [reply _codux/boards/ui-kits/ui-kit-components.board.tsx:42]: the title of the card is below the card - not above
(not my decision)

File: _codux/boards/components/cart/cart-item.board.tsx
Comment (VladyslavG) [inline _codux/boards/components/cart/cart-item.board.tsx:1]: the name of file does not contain `view`

File: src/components/cart/cart-item/cart-item.tsx [hunk 5]
```diff
@@ -15,67 +10,30 @@ export interface CartItemProps {
...
         }
         const newQuantity = parseInt(e.target.value, 10);
         if (newQuantity > 0) {
             updateQuantity({ id: cartItem._id, quantity: newQuantity });
         }
-    }
+    };
 
-    const isAvailable = isCartItemAvailable(cartItem);
+    const handleRemoveItem = () => {
```
Comment (VladyslavG) [inline src/components/cart/cart-item/cart-item.tsx:26]: can be 1 liner

const handleRemoveItem = () =>removeItem(cartItem._id!)

File: src/components/cart/cart-item/cart-item.tsx [hunk 6]
```diff
@@ -15,67 +10,30 @@ export interface CartItemProps {
...
 
 export const CartItem = ({ cartItem, className, isLast }: CartItemProps) => {
-    const name = cartItem.productName?.translated || '';
-    const imageUrl = getImageHttpUrl(cartItem.image, 120, 120);
-
     const { trigger: updateQuantity } = useUpdateCartItemQuantity();
     const { trigger: removeItem } = useRemoveItemFromCart();
 
-    function updateQuantityHandler(e: ChangeEvent<HTMLInputElement>) {
+    const handleQuantityChange = (e: ChangeEvent<HTMLInputElement>) => {
```
Comment (VladyslavG) [inline src/components/cart/cart-item/cart-item.tsx:16]: can be `onQuantityChange` :)

File: src/components/cart/cart-item/cart-item-view.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
+import { Cross2Icon } from '@radix-ui/react-icons';
+import { cart } from '@wix/ecom';
```
Comment (VladyslavG) [inline src/components/cart/cart-item/cart-item-view.tsx]: shouldn't this be type import ?

File: src/components/cart/cart-item/cart-item-view.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+export const CartItemView = ({
+    cartItem,
+    className,
+    isLast,
+    onRemoveButtonClick,
+    onQuantityChange,
+}: CartItemViewProps) => {
+    const name = cartItem.productName?.translated || '';
```
Comment (VladyslavG) [inline src/components/cart/cart-item/cart-item-view.tsx]: || -> ??

File: src/components/cart/cart-item/cart-item-view.tsx [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    const imageUrl = getImageHttpUrl(cartItem.image, 120, 120);
```
Comment (VladyslavG) [inline src/components/cart/cart-item/cart-item-view.tsx]: is it worth moving 120 into a constant ?

File: src/components/cart/cart-item/cart-item.tsx (hunk 5 above)
Comment (yurii-ve) [reply src/components/cart/cart-item/cart-item.tsx:26]: hate oneliners

File: src/components/cart/cart-item/cart-item-view.tsx [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+    return (
+        <div
+            className={classNames(
+                styles.root,
+                { [styles.divider]: !isLast, [styles.outOfStock]: !isAvailable },
+                className
+            )}
+        >
+            <img src={imageUrl} alt={name || ''} className={styles.image} />
```
Comment (VladyslavG) [inline src/components/cart/cart-item/cart-item-view.tsx]: || -> ??

File: src/components/cart/cart-item/cart-item.tsx (hunk 5 above)
Comment (VladyslavG) [reply src/components/cart/cart-item/cart-item.tsx:26]: you just don't know how to cook them

File: src/components/cart/cart-item/cart-item.tsx (hunk 6 above)
Comment (yurii-ve) [reply src/components/cart/cart-item/cart-item.tsx:16]: I like `handleQuantityChange` more

File: src/components/cart/cart-item/cart-item.tsx (hunk 5 above)
Comment (VladyslavG) [reply src/components/cart/cart-item/cart-item.tsx:26]: images

File: src/components/cart/cart-item/cart-item-view.tsx [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+export const CartItemView = ({
+    cartItem,
+    className,
+    isLast,
+    onRemoveButtonClick,
+    onQuantityChange,
+}: CartItemViewProps) => {
+    const name = cartItem.productName?.translated ?? '';
+    const imageUrl = getImageHttpUrl(cartItem.image, IMAGE_SIZE, IMAGE_SIZE);
```
Comment (VladyslavG) [inline src/components/cart/cart-item/cart-item-view.tsx:26]: now it looks horrible, let's return 120 back inline :)

Comment (VladyslavG) [reply src/components/cart/cart-item/cart-item-view.tsx:26]: joking

**Review by VladyslavG — APPROVED** [review]

**Pull request merged**